crmsh/ui_sbd.py

```diff
@@ -0,0 +1,526 @@
+
+    def _configure_show(self, args) -> None:
+        if len(args) > 2:
+            raise self.SyntaxError("Invalid argument")
+        elif len(args) == 2:
+            match args[1]:
+                case "disk_metadata":
+                    self._show_disk_metadata()
+                case "sysconfig":
+                    SBD._show_sysconfig()
+                case "property":
+                    self._show_property()
+                case _:
+                    raise self.SyntaxError(f"Unknown argument: {args[1]}")
+        else:
+            self._show_disk_metadata()
+            if self.device_list_from_config:
+                print()
+            SBD._show_sysconfig()
+            print()
+            self._show_property()
+
+    def _parse_args(self, args: tuple[str, ...]) -> dict[str, int|str]:
+        '''
+        Parse arguments and verify them
+        '''
+        parameter_dict = {}
+
+        for arg in args:
+            match = self.PARSE_RE.match(arg)
+            if not match:
+                raise self.SyntaxError(f"Invalid argument: {arg}")
+            key, suffix, value = match.groups()
+            # timeout related parameters
+            if key in self.TIMEOUT_TYPES and suffix and suffix == "timeout":
+                if not value.isdigit():
+                    raise self.SyntaxError(f"Invalid timeout value: {value}")
+                parameter_dict[key] = int(value)
+            # watchdog device parameter
+            elif key == "watchdog" and suffix == "device":
+                parameter_dict["watchdog-device"] = value
+            else:
+                raise self.SyntaxError(f"Unknown argument: {arg}")
+
+        watchdog_device = parameter_dict.get("watchdog-device")
+        parameter_dict["watchdog-device"] = watchdog.Watchdog.get_watchdog_device(watchdog_device)
+
+        logger.debug("Parsed arguments: %s", parameter_dict)
+        return parameter_dict
+
+    @staticmethod
+    def _adjust_timeout_dict(timeout_dict: dict, diskless: bool = False) -> dict:
+        watchdog_timeout = timeout_dict.get("watchdog")
+        if not watchdog_timeout:
+            watchdog_timeout, _ = sbd.SBDTimeout.get_advised_sbd_timeout(diskless)
+            logger.info("No watchdog timeout specified, use advised value: %s", watchdog_timeout)
+            timeout_dict["watchdog"] = watchdog_timeout
+
+        if diskless:
+            return timeout_dict
+
+        msgwait_timeout = timeout_dict.get("msgwait")
+        if not msgwait_timeout:
+            msgwait_timeout = 2*watchdog_timeout
+            logger.info("No msgwait timeout specified, use 2*watchdog timeout: %s", msgwait_timeout)
+            timeout_dict["msgwait"] = msgwait_timeout
+
+        if msgwait_timeout < 2*watchdog_timeout:
+            logger.warning("It's recommended to set msgwait timeout >= 2*watchdog timeout")
+
+        return timeout_dict
+
+    def _configure_diskbase(self, parameter_dict: dict):
```

Comment:
Ideally, I expect to see msgwait, SBD_DELAY_START, and TimeoutStartSec get updated along with `crm sbd configure watchdog-timeout=50`, for example

Reply:
```
# crm sbd configure show
INFO: crm sbd configure show disk_metadata
==Dumping header on disk /dev/sda5
Header version     : 2.1
UUID               : bf9db259-b5b8-408b-b438-8b229f468107
Number of slots    : 255
Sector size        : 512
Timeout (watchdog) : 15
Timeout (allocate) : 2
Timeout (loop)     : 1
Timeout (msgwait)  : 30
==Header on disk /dev/sda5 is dumped


INFO: crm sbd configure show sysconfig
SBD_PACEMAKER=yes
SBD_STARTMODE=always
SBD_DELAY_START=71
SBD_WATCHDOG_DEV=/dev/watchdog0
SBD_WATCHDOG_TIMEOUT=15
SBD_TIMEOUT_ACTION=flush,reboot
SBD_MOVE_TO_ROOT_CGROUP=auto
SBD_SYNC_RESOURCE_STARTUP=yes
SBD_OPTS=
SBD_DEVICE=/dev/sda5

INFO: crm sbd configure show property
pcmk_delay_max=30s
have-watchdog=true
stonith-enabled=true
stonith-timeout=83
priority-fencing-delay=60

INFO: systemctl show -p TimeoutStartUSec sbd.service --value
TimeoutStartUSec=90

# crm sbd configure watchdog-timeout=50
INFO: No msgwait timeout specified, use 2*watchdog timeout: 100
INFO: Configuring disk-based SBD
INFO: Initializing SBD device /dev/sda5
INFO: Update SBD_WATCHDOG_TIMEOUT in /etc/sysconfig/sbd: 50
INFO: Already synced /etc/sysconfig/sbd to all nodes
WARNING: Resource is running, need to restart cluster service manually on each node
INFO: Update SBD_DELAY_START in /etc/sysconfig/sbd: 141
INFO: Already synced /etc/sysconfig/sbd to all nodes
WARNING: "stonith-timeout" in crm_config is set to 167, it was 83

# crm sbd configure show
INFO: crm sbd configure show disk_metadata
==Dumping header on disk /dev/sda5
Header version     : 2.1
UUID               : 6a296240-c820-4b13-b174-238b9e49e6d9
Number of slots    : 255
Sector size        : 512
Timeout (watchdog) : 50
Timeout (allocate) : 2
Timeout (loop)     : 1
Timeout (msgwait)  : 100
==Header on disk /dev/sda5 is dumped


INFO: crm sbd configure show sysconfig
SBD_PACEMAKER=yes
SBD_STARTMODE=always
SBD_DELAY_START=141
SBD_WATCHDOG_DEV=/dev/watchdog0
SBD_WATCHDOG_TIMEOUT=50
SBD_TIMEOUT_ACTION=flush,reboot
SBD_MOVE_TO_ROOT_CGROUP=auto
SBD_SYNC_RESOURCE_STARTUP=yes
SBD_OPTS=
SBD_DEVICE=/dev/sda5

INFO: crm sbd configure show property
pcmk_delay_max=30s
have-watchdog=true
stonith-enabled=true
stonith-timeout=167
priority-fencing-delay=60

INFO: systemctl show -p TimeoutStartUSec sbd.service --value
TimeoutStartUSec=169
```